src/app/myworkflows/my-workflow/my-workflow.component.ts

```diff
@@ -81,44 +92,95 @@ export class MyWorkflowComponent extends MyEntry implements OnInit {
         }
       }
     );
-    this.userService.user$.subscribe(user => {
+
+    // Retrieve all of the workflows for the user and update the workflow service
+    this.userService.user$.pipe(takeUntil(this.ngUnsubscribe)).subscribe(user => {
       if (user) {
         this.user = user;
-        this.usersService.userWorkflows(user.id).pipe(first()).subscribe(workflows => {
-          this.workflowService.setWorkflows(workflows);
-        });
-      }
-    });
-    this.workflowService.workflows$.pipe(takeUntil(this.ngUnsubscribe)).subscribe(workflows => {
-      if (workflows) {
-        this.workflows = workflows;
-        const sortedWorkflows = this.myworkflowService.sortGroupEntries(workflows, this.user.username, 'workflow');
-        this.selectInitialEntry(sortedWorkflows);
+        combineLatest(this.usersService.userWorkflows(user.id).pipe(first()), this.workflowsService.sharedWorkflows().pipe(first()))
+          .subscribe(([workflows, sharedWorkflows]) => {
+            if (workflows && sharedWorkflows) {
+              this.workflowService.setWorkflows(workflows);
+              this.workflowService.setSharedWorkflows(sharedWorkflows);
+            }
+          });
       }
     });
+
+    // Using the workflows and shared with me workflows, initialize the organization groupings and set the initial entry
+    combineLatest(this.workflowService.workflows$.pipe(takeUntil(this.ngUnsubscribe)),
+      this.workflowService.sharedWorkflows$.pipe(takeUntil(this.ngUnsubscribe)))
+      .subscribe(([workflows, sharedWorkflows]) => {
+        if (workflows && sharedWorkflows) {
+          this.workflows = workflows;
+          const sortedWorkflows = this.myworkflowService.sortGroupEntries(workflows, this.user.username, 'workflow');
+          this.setGroupEntriesObject(sortedWorkflows);
+
+          this.sharedWorkflows = sharedWorkflows;
+          const sortedSharedWorkflows = this.myworkflowService.sortGroupEntries(sharedWorkflows, this.user.username, 'workflow');
+          this.setGroupSharedEntriesObject(sortedSharedWorkflows);
+
+          if (!this.hasLoadedWorkflows) {
+            if (this.workflows.length > 0) {
+              this.selectInitialEntry(sortedWorkflows);
+            } else if (this.sharedWorkflows.length > 0) {
+              this.selectInitialEntry(sortedSharedWorkflows);
+            }
+          }
+        }
+      });
+
     this.stateService.refreshMessage$.subscribe(refreshMessage => this.refreshMessage = refreshMessage);
   }
 
+  /**
+   * Sets the sorted entries for display in dropdowns
+   * @param sortedEntries Array of sorted entries
+   */
+  public setGroupSharedEntriesObject(sortedEntries: any): void {
```

Comment:
There is a lot of any in the existing code, and this PR involved somewhat duplicating some of the functions and moving stuff around, which explains why there was so many anys appearing as new code. I'll go through the changed code again and clean up the anys that I can.

Reply:
Sounds good. BTW, the reason I requested changes instead of approving was for the isUserOwner() in workflow.component.html -- this one was a nice to have but not a blocker.